tools/lsp/common/rename_component.rs

```diff
@@ -0,0 +1,781 @@
+        for e in document_node.ExportsList() {
+            if let Some(c) = e.Component() {
+                let cid = c.DeclaredIdentifier();
+                if cid.text().to_string().trim() == name {
+                    return Some(cid);
+                }
+            }
+        }
+        None
+    }
+
+    #[test]
+    fn test_rename_component_from_definition_ok() {
+        let component_instance = test::compile_test_with_sources(
+            "fluent",
+            HashMap::from([(
+                test::main_test_file_name(),
+                r#"
+component Foo { }
+
+component Baz {
+    Foo { }
+}
+
+export component Bar {
+    Foo { }
+    Rectangle {
+        Foo { }
+        Baz { }
+    }
+
+    if true: Rectangle {
+        Foo { }
+    }
+
+    for i in [1, 2, 3]: Foo { }
+}
+                    "#
+                .to_string(),
+            )]),
+        );
+
+        let tl = component_instance.definition().type_loader();
+        let doc = tl.get_document(&test::main_test_file_name()).unwrap();
+
+        let foo_identifier =
+            find_component_declared_identifier(doc.node.as_ref().unwrap(), "Foo").unwrap();
+        let edit = rename_component_from_definition(&tl, &foo_identifier, "XxxYyyZzz".to_string())
+            .unwrap();
+
+        let edited_text = compile_test_changes(&component_instance, &edit);
+
+        assert_eq!(edited_text.len(), 1);
+        assert!(edited_text[0].contents.contains("XxxYyyZzz"));
+        assert!(!edited_text[0].contents.contains("Foo"));
+    }
+
+    #[test]
+    fn test_rename_component_from_definition_with_renaming_export_ok() {
+        let component_instance = test::compile_test_with_sources(
+            "fluent",
+            HashMap::from([
+                (
+                    test::main_test_file_name(),
+                    r#"
+import { FExport} from "source.slint";
+
+export component Foo {
+    FExport { }
+}
+                "#
+                    .to_string(),
+                ),
+                (
+                    PathBuf::from("/source.slint"),
+                    r#"
+component Foo { }
+
+export { Foo as FExport }
+                "#
+                    .to_string(),
+                ),
+            ]),
+        );
+
+        let tl = component_instance.definition().type_loader();
+        let doc = tl.get_document(&PathBuf::from("/source.slint")).unwrap();
+
+        let foo_identifier =
+            find_component_declared_identifier(doc.node.as_ref().unwrap(), "Foo").unwrap();
+        let edit = rename_component_from_definition(&tl, &foo_identifier, "XxxYyyZzz".to_string())
+            .unwrap();
+
+        let edited_text = compile_test_changes(&component_instance, &edit);
+
+        assert_eq!(edited_text.len(), 1);
+        assert_eq!(edited_text[0].url.to_file_path().unwrap().to_string_lossy(), "/source.slint");
+        assert!(edited_text[0].contents.contains("XxxYyyZzz"));
+        assert!(!edited_text[0].contents.contains("Foo"));
+    }
```

Comment:
the tests could also check that that the modified source is equal to some expected value.
I think that'd made the test easier to understand.